yellow-paper/docs/gas-and-fees/gas-metering-and-fees.md

```diff
@@ -0,0 +1,683 @@
+    a2 --> BaseRollup
+    a3 --> BaseRollup
+    BaseRollup  --> [*]
+    }
+```
+
+# The Aztec token
+
+:::warning
+This token will not be called the Aztec token in the final implementation. It, and the symbol `$AZT`, are placeholders.
+:::
+
+`$AZT` is an enshrined token that is used to pay fees on Aztec. It has the following properties:
+
+- It is fungible
+- It is obtained via a "1-way bridge" from Ethereum
+- Its balances are public
+- It cannot be transferred between users
+- It exposes a public `pay_fee` function which inspects the public context for the `transaction_fee`, and reduces the caller's balance by that amount
+
+
+# Transaction costs
+
+### Transaction specific costs
+
+Transactions specify/consume:
+- variable amounts "gas", which must be multiplied by a "fee per gas" to arrive at a fee in `$AZT`.
+- fixed amounts of `$AZT`
+
+Transactions consume gas across three dimensions:
+
+- making the data associated with the transaction available
+- computation on L2 
+- sending messages to L1
+
+We meter each of these dimensions per transaction as "da gas", "l2 gas", and "l1 gas", respectively.
+
+### Transaction overhead costs
+
+There are costs associated with including transactions in blocks, and blocks in the rollup, independent of what the transaction does.
+
+These include: 
+
+- verification of transaction's private kernel proof
+- verification of nullifiers
+- proving the rollup circuits
+- verification of the root rollup proof on L1
+
+For any transaction in a block of a given size at a given time, the costs are the same.
+
+Since these costs are paid by actors *other than* the user who submits the transaction (e.g. sequencers), the user must be able to specify a fee that they are willing to pay for these costs for inclusion.
+
+We term this a transaction's "inclusion fee", as it is the flat fee (in `$AZT`) the user is willing to pay for the transaction to be included in a block.
+
+:::note Overhead costs are not constant
+The costs the inclusion fee is thought to cover are **not constant**.
+
+E.g. if an identical transaction is submitted at time `t` and time `t+1`, the costs to the sequencer might be higher at `t+1` because the the gas price on L1 has increased, and thus verification costs are higher.
+
+We will need to develop a mechanism to ensure the user can specify a fee that is both reasonable and likely to be included in a block, and that the sequencer is compensated fairly for the costs they incur.
+
+Open discussions are being had around using a [Shapley cost sharing mechanism here](https://en.wikipedia.org/wiki/Shapley_value).
+
+In the immediate term, we will risk bad UX: the user will need to look at the `inclusion_fee` from previously included transactions to get a sense of what is reasonable, and thus likely overpay.
+:::
+
+# Fees per gas
+
+The fees per unit gas are set by the protocol, and are not part of the user's transaction.
+
+The fees are:
```

Comment:
I'm not really sure about this. This feels very 1559-ish, which I think it is far from certain that we plan to do.